common/src/chain/block/timestamp.rs

```diff
@@ -54,6 +54,14 @@ impl BlockTimestamp {
     }
 }
 
+impl From<Duration> for BlockTimestamp {
```

Comment:
Please remove this conversion and use `from_duration_since_epoch()` instead. This can be ambiguous as the starting time isn't clear.

Reply:
I believe `from_duration_since_epoch` is misleading because nothing prevents me from passing any `Duration` unrelated to the epoch. I was thinking about removing these 2 functions leaving only `From` conversion as a universal one.

Reply:
ok, reverted